From 34a1e7ebe92903351a143dbdfbb2deee35abb869 Mon Sep 17 00:00:00 2001
From: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Date: Thu, 3 Feb 2022 16:41:51 +0200
Subject: [PATCH 03/43] s32gen1: remove sram dts nodes and memory ranges

Without the sram dts nodes there is no need to perform any special
operation on the SRAM memory ranges. Remove the nodes and the code
related to altering SRAM memory ranges.

Issue: ALB-8350
Upstream-Status: Pending 

Signed-off-by: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/cpu.c   | 20 +-------------------
 arch/arm/cpu/armv8/s32/fdt.c   | 16 ----------------
 arch/arm/dts/fsl-s32g274a.dtsi |  5 -----
 arch/arm/dts/fsl-s32g3.dtsi    |  5 -----
 arch/arm/dts/fsl-s32r45.dts    |  5 -----
 5 files changed, 1 insertion(+), 50 deletions(-)

diff --git a/arch/arm/cpu/armv8/s32/cpu.c b/arch/arm/cpu/armv8/s32/cpu.c
index 345ec369ed..1c4a492558 100644
--- a/arch/arm/cpu/armv8/s32/cpu.c
+++ b/arch/arm/cpu/armv8/s32/cpu.c
@@ -407,10 +407,6 @@ static void s32_init_ram_size(void)
 		start = gd->bd->bi_dram[i].start;
 		size = gd->bd->bi_dram[i].size;
 
-		/* Don't advertise SRAM */
-		if (start == S32_SRAM_BASE)
-			continue;
-
 		if (!start && !size)
 			continue;
 
@@ -420,22 +416,12 @@ static void s32_init_ram_size(void)
 
 int dram_init_banksize(void)
 {
-#if defined(CONFIG_S32_SKIP_RELOC) && !defined(CONFIG_S32_ATF_BOOT_FLOW)
-	gd->bd->bi_dram[0].start = S32_SRAM_BASE;
-	gd->bd->bi_dram[0].size = get_sram_size();
-
-	gd->bd->bi_dram[1].start = 0x0;
-	gd->bd->bi_dram[1].size = 0x0;
-
-	gd->bd->bi_dram[2].start = 0x0;
-	gd->bd->bi_dram[2].size = 0x0;
-#else
 	int ret;
 
 	ret = fdtdec_setup_memory_banksize();
 	if (ret)
 		return ret;
-#endif
+
 	s32_init_ram_size();
 
 	return 0;
@@ -451,11 +437,7 @@ phys_size_t __weak get_effective_memsize(void)
 	 * Restrict U-Boot area to the first bank of the DDR memory.
 	 * Note: gd->bd isn't initialized yet
 	 */
-#if defined(CONFIG_S32_SKIP_RELOC) && !defined(CONFIG_S32_ATF_BOOT_FLOW)
-	size = get_sram_size();
-#else
 	size = CONFIG_SYS_FSL_DRAM_SIZE1;
-#endif
 
 	/* Get first DDR bank size from DT 'memory' node */
 	while ((nodeoff = fdt_node_offset_by_prop_value(gd->fdt_blob, nodeoff,
diff --git a/arch/arm/cpu/armv8/s32/fdt.c b/arch/arm/cpu/armv8/s32/fdt.c
index 340e66e721..b0c66d477f 100644
--- a/arch/arm/cpu/armv8/s32/fdt.c
+++ b/arch/arm/cpu/armv8/s32/fdt.c
@@ -76,19 +76,6 @@ static void ft_fixup_ddr_polling(const void *old_blob, void *new_blob)
 		       fdt_strerror(ret));
 }
 
-static void hide_sram(bd_t *bd)
-{
-	int bank;
-
-	for (bank = 0; bank < CONFIG_NR_DRAM_BANKS; bank++) {
-		if (bd->bi_dram[bank].start == S32_SRAM_BASE) {
-			bd->bi_dram[bank].start = 0;
-			bd->bi_dram[bank].size = 0;
-			break;
-		}
-	}
-}
-
 static void apply_memory_fixups(void *blob, bd_t *bd)
 {
 	u64 start[CONFIG_NR_DRAM_BANKS];
@@ -141,12 +128,9 @@ static void apply_ddr_limits(bd_t *bd)
 
 static void ft_fixup_memory(void *blob, bd_t *bd)
 {
-	hide_sram(bd);
-
 	apply_ddr_limits(bd);
 
 	apply_memory_fixups(blob, bd);
-
 }
 
 #ifdef CONFIG_S32_ATF_BOOT_FLOW
diff --git a/arch/arm/dts/fsl-s32g274a.dtsi b/arch/arm/dts/fsl-s32g274a.dtsi
index 58ecc03280..3f4a8291f2 100644
--- a/arch/arm/dts/fsl-s32g274a.dtsi
+++ b/arch/arm/dts/fsl-s32g274a.dtsi
@@ -23,11 +23,6 @@
 		device_type = "memory";
 		reg = <0x8 0x80000000 0 0x80000000>;
 	};
-
-	sram@34000000 {
-		device_type = "memory";
-		reg = <0 0x34000000 0 0x800000>;
-	};
 };
 
 &siul2_1_nvram {
diff --git a/arch/arm/dts/fsl-s32g3.dtsi b/arch/arm/dts/fsl-s32g3.dtsi
index 4d641e9c10..c46df35f02 100644
--- a/arch/arm/dts/fsl-s32g3.dtsi
+++ b/arch/arm/dts/fsl-s32g3.dtsi
@@ -25,11 +25,6 @@
 		reg = <0x8 0x80000000 0 0x80000000>;
 	};
 
-	sram@34000000 {
-		device_type = "memory";
-		reg = <0 0x34000000 0 0x1400000>;
-	};
-
 	gic: interrupt-controller@50800000 {
 		compatible = "arm,gic-v3";
 		#interrupt-cells = <3>;
diff --git a/arch/arm/dts/fsl-s32r45.dts b/arch/arm/dts/fsl-s32r45.dts
index 3d1e4eca25..62f59d23f5 100644
--- a/arch/arm/dts/fsl-s32r45.dts
+++ b/arch/arm/dts/fsl-s32r45.dts
@@ -33,11 +33,6 @@
 		reg = <0x8 0x80000000 0 0x80000000>;
 	};
 
-	sram@34000000 {
-		device_type = "memory";
-		reg = <0 0x34000000 0 0x800000>;
-	};
-
 	signature {
 		key-boot_key {
 			required = "conf";
-- 
2.17.1

